### PR TITLE
docs: fix bugs in contribution, getting started guides

### PR DESCRIPTION
**Pull request by RikkiButler20**
Making sure the documentation is up to date

### CODE REVIEW

**Review by tgamblin — APPROVED** [review]
Thanks!

File: lib/spack/docs/getting_started.rst [hunk 1]
```diff
@@ -54,8 +54,7 @@ to your path and you're ready to go:
 
 .. code-block:: console
 
-   $ export PATH=$SPACK_ROOT/bin:$PATH
-   $ spack install libelf
+   $ . share/spack/setup-env.sh
```
Comment (becker33) [inline lib/spack/docs/getting_started.rst]: I think you probably want to keep the `spack install libelf` line.
```suggestion
   $ . share/spack/setup-env.sh 
   $ spack install libelf
```

**Review by becker33 — CHANGES_REQUESTED** [review]

**Review by becker33 — APPROVED** [review]

File: lib/spack/docs/getting_started.rst [hunk 2]
```diff
@@ -54,7 +54,7 @@ to your path and you're ready to go:
 
 .. code-block:: console
 
-   $ export PATH=$SPACK_ROOT/bin:$PATH
+   $ . share/spack/setup-env.sh
```
Comment (adamjstewart) [inline lib/spack/docs/getting_started.rst]: I think you should undo this change. The paragraph above mentions adding Spack to the `PATH`, which is what we're doing here. The paragraph below mentions using Spack's shell support, which is duplicated by your change.

File: lib/spack/docs/getting_started.rst [hunk 3]
```diff
@@ -53,6 +53,13 @@ in the ``SPACK_ROOT`` environment variable.  Add ``$SPACK_ROOT/bin``
...
 .. code-block:: console
+   
+   # For bash/zsh users
+   $ export SPACK_ROOT=/path/to/spack
+
+   # For tcsh or csh users (note you must set SPACK_ROOT)
+   $ setenv SPACK_ROOT /path/to/spack
+   
 
    $ export PATH=$SPACK_ROOT/bin:$PATH
```
Comment (adamjstewart) [inline lib/spack/docs/getting_started.rst]: This line is bash/zsh specific. We should also add a line for fish now that we support fish.

File: lib/spack/docs/getting_started.rst [hunk 4]
```diff
@@ -53,6 +53,15 @@ in the ``SPACK_ROOT`` environment variable.  Add ``$SPACK_ROOT/bin``
...
 .. code-block:: console
+   
+   # For bash/zsh users
+   $ export SPACK_ROOT=/path/to/spack
+
+   # For tcsh or csh users (note you must set SPACK_ROOT)
+   $ setenv SPACK_ROOT /path/to/spack
+
+   #For fish users
+   $ set SPACK_ROOT /path/to/spack
```
Comment (becker33) [inline lib/spack/docs/getting_started.rst]:
```suggestion
   $ set -x SPACK_ROOT /path/to/spack
```

**Review by becker33 — CHANGES_REQUESTED** [review]

File: lib/spack/docs/getting_started.rst (hunk 3 above)
Comment (RikkiButler20) [reply lib/spack/docs/getting_started.rst]: Added in the lines for fish users

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@RikkiButler20 Feel free to ping me whenever you think this is ready to be merged.

File: lib/spack/docs/getting_started.rst (hunk 3 above)
Comment (adamjstewart) [reply lib/spack/docs/getting_started.rst]: This line is still bash/zsh specific

File: lib/spack/docs/getting_started.rst [hunk 5]
```diff
@@ -53,6 +53,16 @@ in the ``SPACK_ROOT`` environment variable.  Add ``$SPACK_ROOT/bin``
...
 
 .. code-block:: console
+   
+   # For bash/zsh users
+   $ export SPACK_ROOT=/path/to/spack
+
+   # For tcsh or csh users (note you must set SPACK_ROOT)
+   $ setenv SPACK_ROOT /path/to/spack
+
+   #For fish users
```
Comment (adamjstewart) [inline lib/spack/docs/getting_started.rst]:
```suggestion
   # For fish users
```

File: lib/spack/docs/getting_started.rst (hunk 3 above)
Comment (RikkiButler20) [reply lib/spack/docs/getting_started.rst]: I've added the line for fish

Comment (adamjstewart) [reply lib/spack/docs/getting_started.rst]: The line for fish has been added, but the `export PATH=` line only works on bash/zsh. We should move this up and add the equivalent logic for other shells. Or we could just add the logic for `PATH` and ignore `SPACK_ROOT`, assuming it is already known to the user. That's what the existing docs were trying to get at.

File: lib/spack/docs/getting_started.rst [hunk 6]
```diff
@@ -53,22 +53,38 @@ in the ``SPACK_ROOT`` environment variable.  Add ``$SPACK_ROOT/bin``
...
 
 .. code-block:: console
+   
+   # For bash/zsh users
+   $ export SPACK_ROOT=/path/to/spack
+   $ export PATH=$SPACK_ROOT/bin:$PATH
 
+   # For tsch/csh users 
+   $ setenv SPACK_ROOT /path/to/spack
    $ export PATH=$SPACK_ROOT/bin:$PATH
```
Comment (adamjstewart) [inline lib/spack/docs/getting_started.rst]:
```suggestion
   $ setenv PATH $SPACK_ROOT/bin:$PATH
```

File: lib/spack/docs/getting_started.rst [hunk 7]
```diff
@@ -53,22 +53,37 @@ in the ``SPACK_ROOT`` environment variable.  Add ``$SPACK_ROOT/bin``
...
+
+.. code-block:: console
+
    $ spack install libelf
 
 For a richer experience, use Spack's shell support:
 
 .. code-block:: console
 
+   #Note you must set SPACK_ROOT
```
Comment (tldahlgren) [inline lib/spack/docs/getting_started.rst]: LGTM.  I do suggest adding a space between the hash `#` and `Note`.

Comment (tldahlgren) [reply lib/spack/docs/getting_started.rst]: But it is not a blocker since you already have 2 approvals.

**Review by becker33 — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Thanks!
